synthesizer/process/src/trace/mod.rs

```diff
@@ -282,15 +276,19 @@ impl<N: Network> Trace<N> {
             batch_inclusions.push(assignment.to_circuit_assignment::<A>()?);
         }
 
+        // Fetch the inclusion proving key.
+        let inclusion_proving_key = ProvingKey::<N>::new(N::inclusion_proving_key().clone());
+
```

Comment:
```suggestion
        // Fetch the inclusion proving key.
        let inclusion_proving_key = ProvingKey::<N>::new(N::inclusion_proving_key().clone());

```

This should be moved back into the `if` statement. Otherwise, we are needlessly loading this proving key.

Reply:
I'm defining it here to make sure it has the appropriate lifetime:
- we are inserting it in line 284 by **reference**.
- Which I'm doing because the proving keys are inserted by reference in line 265.
- Which I'm doing because `&self.transition_tasks` is directly passed by reference in line 165. 

Before this PR we made copies of the `transition_tasks`, which was fine for the provingkeys (which were Arcs) but not fine for the Assignments. Maybe there is a slightly cleaner way but it does not seem worth the time to refactor.